semanticdb/semanticdb3/semanticdb3.proto

```diff
@@ -0,0 +1,96 @@
+syntax = "proto3";
+
+package scala.meta.internal.semanticdb3;
+
+message TextDocuments {
+  repeated TextDocument documents = 1;
+}
+
+message TextDocument {
+  string format = 1;
+  string uri = 2;
+  string text = 3;
+  string language = 4;
+  repeated SymbolInformation symbols = 5;
+  repeated SymbolOccurrence occurrences = 6;
+  repeated Diagnostic diagnostics = 7;
+  repeated Synthetic synthetics = 8;
+}
+
+message Range {
+  int32 start_line = 1;
+  int32 start_character = 2;
+  int32 end_line = 3;
+  int32 end_character = 4;
```

Comment:
Based on the feedback in https://github.com/scalameta/scalameta/pull/1220, this is the only controversial change in the pullrequest:
  * Previous discussion: https://github.com/scalameta/scalameta/pull/1220#discussion_r162564442
  * Performance results: https://github.com/scalameta/scalameta/pull/1220#discussion_r164602743